### ISSUE TITLE
Filter events by categories

### ISSUE THREAD

**Issue by sarahsporck** [digitalfabrik/integreat-app#2707]
**Is your feature request related to a problem? Please describe.**
I'm looking for all fun & leisure events in IG, but because of thousand events I cannot find those. It would be so nice to be able to filter events by category (maybe reuse the poi categories?)

**Describe the solution you'd like**
Implement a filter for web & native for categories. The filter should look nice together with #2706

**Describe alternatives you've considered**
N/A

**Additional context**
Relates: #2706 
CMS issue: https://github.com/digitalfabrik/integreat-cms/issues/273

**Related issues**
#2986, #2935, #2808, #2112

**Comment by hauf-toni** [comment]
Design can be found [📍 here](https://www.figma.com/file/cA4F2MwHs2LNGviOWUjfE9/Integreat-Tickets-Frontend?type=design&node-id=2685%3A2558&mode=design&t=Y8dWJzbl4UeGTy6G-1)

**Comment by JoeyStk** [comment]
@steffenkleinle we weren't aware that you were waiting for us on this topic. Should we schedule this topic for us in the near future? :)

**Comment by steffenkleinle** [comment]
> [@steffenkleinle](https://github.com/steffenkleinle) we weren't aware that you were waiting for us on this topic. Should we schedule this topic for us in the near future? :)

@JoeyStk thanks for checkig in! I think we are in no rush, so we are fine with this taking another few months :) In the end a decision for @hauf-toni / @osmers I guess :)
Just to make sure: Are the categories in the ticket finally decided on or should there still be some discussion on which categories to use?

**Comment by JoeyStk** [comment]
> Just to make sure: Are the categories in the ticket finally decided on or should there still be some discussion on which categories to use?

I don't know. I'll also leave this for @hauf-toni and @osmers to decide. I'll add another question to this: Should the event categories be the same as POI categories, thus should they be created by the service team or can they be hardcoded by us?

**Comment by laravoelk** [comment]
I have just talked to @osmers and @hauf-toni and we will discuss different event categories in the team and will get back to you. :-)

**Comment by laravoelk** [comment]
Sorry that it took a while. Here are the categories, that we find very important:

- Location (Ort) - especially important in Landkreisen to filter for the single locations (Orte)
- Kids & Family
- Sport
- Health
- Women
- Senior citizens
- Language
- Leisure
- Politics
- Arts & Culture
- Integration
- Markets
- Religion
- Career and further education
- Consulting services
- Others

**Comment by steffenkleinle** [comment]
I think that sounds great, thank you. Just a few questions:

> Location (Ort) - especially important in Landkreisen to filter for the single locations (Orte)

@laravoelk can you elaborate a little bit more what you mean by that please? I sadly don't understand this.

> Markets

What is meant by `Markets`?

@hauf-toni can you have a look at this please? Do you think this would work with our current design approach?

Does this correspond with our current poi/location categories? I think it would be nice to have them as similar as possible.

**Comment by laravoelk** [comment]
Sure: Districts (Landkreise) include many different communities (Gemeinden) and it would therefore be very useful to be able to filter where (meaning in which district) the event will take place. As this is obviously never the same, I am not sure how this can be done technically. :-)

By markets we mean flea markets, farmers markets etc.

Does that answer your questions? :-) Otherwise just let me know.

**Comment by steffenkleinle** [comment]
> Sure: Districts (Landkreise) include many different communities (Gemeinden) and it would therefore be very useful to be able to filter where (meaning in which district) the event will take place. As this is obviously never the same, I am not sure how this can be done technically. :-)

Thanks for clarifying. I don't think this falls in the same category as everything else and I think this should be a separate issue/task. @hauf-toni let me know if you disagree.

**Comment by hauf-toni** [comment]
Agreed @steffenkleinle – Lara and I just discussed that an alternative approach could be to offer a filter based on the user's location plus a radius of X km, similar to other applications. Providing a separate option for each Gemeinde would likely require significant effort, be challenging to maintain, and could overly complicate the user interface on the front end.

**Comment by steffenkleinle** [comment]
Alright, since the categories have to be implemented in the cms, this is blocked by #https://github.com/digitalfabrik/integreat-cms/issues/273. Please clarify all cms-side questions there :)

**Comment by steffenkleinle** [comment]
> Isn't it more useful to enable every municipality to use their own tags to filter despite the fact that we need to solve the translation problem of those tags?

Well... I'd say depends. If a municipality knows what they are doing, it might be a good thing. However, I am quite afraid that this is not always the case and we'll end up with too many, too specific categories that are inconsistent, also in terms of naming and translations (and perhaps icons, if necessary). I am pretty sure that this will lead to "Wildwuchs" (uncontrolled growth). So from a development/maintenance perspective we should definitely prefer centralized categories imo. 

Furthermore, we should definitely separate categories and locations, from a users perspective those are entirely different things and mixing them makes things confusing and harder to find.

> Users should be able to select multiple categories in the app/web app.

Why? What is the use case to select multiple categories at the same time?

> When multiple selections are made, only events that contain all of the selected categories are displayed.

I want to question whether it makes sense that one event is part of multiple categories. Perhaps you are more thinking of the concept of "tags" here?

> All categories that do not contain “integreat” in their name are displayed in the app/web app (we need this so that the iCal import filter is not displayed if it is used).

I am not sure I understand this, can you please elaborate on this?
